3.7.0-RC1 broken with SN 0.5.9

I wanted to try cats-effect 3.7.0-RC1 against the latest release of Scala Native 0.5.9. Unfortunately producing binary fails with a following error:
```
Compiling project (Scala 3.7.2, Scala Native 0.5.9)
Compiled project (Scala 3.7.2, Scala Native 0.5.9)
[info] Linking (multithreadingEnabled=true, disable if not used) (9700 ms)
[info] Discovered 2666 classes and 17685 methods after classloading
[info] Checking intermediate code (quick) (497 ms)
[info] Discovered 2638 classes and 13835 methods after optimization
[info] Optimizing (debug mode) (20444 ms)
[info] Produced 13 LLVM IR files
[info] Generating intermediate code (14405 ms)
[error] /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:9:22: error: variable has incomplete type 'struct sigaction'
[error]     9 |     struct sigaction action;
[error]       |                      ^
[error] /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:9:12: note: forward declaration of 'struct sigaction'
[error]     9 |     struct sigaction action;
[error]       |            ^
[error] /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:12:13: error: call to undeclared function 'sigemptyset'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
[error]    12 |     error = sigemptyset(&action.sa_mask);
[error]       |             ^
[error] /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:16:13: error: call to undeclared function 'sigaddset'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
[error]    16 |     error = sigaddset(&action.sa_mask, 13); // mask SIGPIPE
[error]       |             ^
[error] /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:20:12: error: call to undeclared function 'sigaction'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
[error]    20 |     return sigaction(signum, &action, NULL);
[error]       |            ^
[error] 4 errors generated.
[info] Compiling to native code (10068 ms)
[info] Total (52793 ms)
Exception in thread "main" scala.scalanative.build.BuildException: Failed to compile /tmp/3168152424178068910/.scala-build/3168152424178068910_16c8be8ad5-2b2bc60792/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c
        at scala.scalanative.build.LLVM$.$anonfun$compileFile$1(LLVM.scala:123)
        at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:42)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:74)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Error: scala.build.errors.ScalaNativeBuildError: Error compiling with Scala Native
```

Here's a minimal gist to reproduce it https://gist.github.com/majk-p/af00a196ab8c193994992bf8ad7d2c89

To reproduce run: `scala run https://gist.github.com/majk-p/af00a196ab8c193994992bf8ad7d2c89`

* Could not replicate using macOS arm, 16.mumble
   `scala-cli  broken_ce.scala`  where the file is the gist above.

* Did replicate on Intel Linux 6.17.3+
```
Optimizing (debug mode) (7824 ms)
[error] /mnt/AthaBHdd1/lee49/DevoHdd1/scala_native/WorkSpace_1/2025/.scala-build/2025_b3557dfd11-dc7b6f9c58/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:9:22: error: variable has incomplete type 'struct sigaction'
[error]     9 |     struct sigaction action;
[error]       |                      ^
[error] /mnt/AthaBHdd1/lee49/DevoHdd1/scala_native/WorkSpace_1/2025/.scala-build/2025_b3557dfd11-dc7b6f9c58/native/native/dependencies/cats-effect_native0.5_3-3.7.0-RC1-1/scala-native/signal_helper.c:9:12: note: forward declaration of 'struct sigaction'
[error]     9 |     struct sigaction action;
[error]       |            ^

// And a bunch more
```

This adds evidence to the `missing .h` file and shows CE `signal_helper.c` as the probably problem child.

Let me know if I can help resolve this.

Some success, thanks to @ekrich & the reproducer above

```
$ scala-cli  --native-compile "-D_GNU_SOURCE" broken_ce.scala
[info] Linking (multithreadingEnabled=true, disable if not used) (5900 ms)
[info] Discovered 2666 classes and 17685 methods after classloading
[info] Checking intermediate code (quick) (288 ms)
[info] Discovered 2638 classes and 13835 methods after optimization
[info] Optimizing (debug mode) (7811 ms)
[error] /mnt/AthaBHdd1/lee49/DevoHdd1/scala_native/WorkSpace_1/2025/.scala-build/2025_b3557dfd11-dc7b6f9c58/native/native/dependencies/javalib_native0.5_3-0.5.9-0/scala-native/sys/linux_sched.c:11:9: warning: '_GNU_SOURCE' macro redefined [-Wmacro-redefined]
[error]    11 | #define _GNU_SOURCE
[error]       |         ^
[error] <command line>:1:9: note: previous definition is here
[error]     1 | #define _GNU_SOURCE 1
[error]       |         ^
[error] 1 warning generated.
[info] Produced 5 LLVM IR files
[info] Generating intermediate code (11084 ms)
[info] Compiling to native code (7308 ms)
[info] Linking with [pthread, dl, m]
[info] Linking native code (immix gc, none lto) (448 ms)
[info] Postprocessing (0 ms)
[info] Total (30318 ms)
Hello!

```

Yes, the redefinition message should probably be fixed for SN 0.5.10-SNAPSHOT
to be guarded by a roughly, not compiled:
```
#ifndef _GNU_SOURCE
# define _GNU_SOURCE 1
#endif
```
I think unconditionally defining it to the same thing it already is would work also,
but guarding it is clearer, at least to me.

Since the file is 'linux_mumble', there is probably not much chance to expunge
the use of _GNU_SOURCE.

A quick fix for CE would be to add that guard code to the top of `signal_helper.c`.

I am off to rat around in linux `signal.h` to figure out why it seems to need _GNU_SOURCE.
There may be a way to eliminate that requirement.

--- a/build.sbt
+++ b/build.sbt
@@ -363,7 +363,7 @@ Global / tlCommandAliases ++= Map(
 
 lazy val nativeTestSettings = Seq(
   nativeConfig ~= { c =>
-    c.withSourceLevelDebuggingConfig(_.enableAll.generateFunctionSourcePositions(false))
+    c.withSourceLevelDebuggingConfig(_.enableAll)
       .withOptimize(
         true
       ) // `false` doesn't work due to https://github.com/scala-native/scala-native/issues/4366

--- a/core/native/src/main/resources/scala-native/signal_helper.c
+++ b/core/native/src/main/resources/scala-native/signal_helper.c
@@ -1,7 +1,9 @@
 #ifdef CATS_EFFECT_SIGNAL_HELPER
 
 // we'll need POSIX for `sigaction`:
+#ifndef _POSIX_C_SOURCE
 #define _POSIX_C_SOURCE 1
+#endif
 
 #include <stddef.h>
 #include <signal.h>

--- a/core/native/src/main/resources/scala-native/signal_helper.c
+++ b/core/native/src/main/resources/scala-native/signal_helper.c
@@ -1,9 +1,9 @@
 #ifdef CATS_EFFECT_SIGNAL_HELPER
 
-#include <stddef.h>
-
 // we'll need POSIX for `sigaction`:
 #define _POSIX_C_SOURCE 1
+
+#include <stddef.h>
 #include <signal.h>
 
 typedef void (*Handler)(int);

--- a/core/native/src/main/resources/scala-native/signal_helper.c
+++ b/core/native/src/main/resources/scala-native/signal_helper.c
@@ -1,5 +1,9 @@
 #ifdef CATS_EFFECT_SIGNAL_HELPER
+
 #include <stddef.h>
+
+// we'll need POSIX for `sigaction`:
+#define _POSIX_C_SOURCE 1
 #include <signal.h>
 
 typedef void (*Handler)(int);
@@ -19,4 +23,5 @@ int cats_effect_install_handler(int signum, Handler handler) {
     }
     return sigaction(signum, &action, NULL);
 }
+
 #endif

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -3,7 +3,7 @@ libraryDependencies += "org.scala-js" %% "scalajs-env-selenium" % "1.1.1"
 addSbtPlugin("org.typelevel" % "sbt-typelevel" % "0.8.0")
 
 addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.18.2")
-addSbtPlugin("org.scala-native" % "sbt-scala-native" % "0.5.8")
+addSbtPlugin("org.scala-native" % "sbt-scala-native" % "0.5.9")
 addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.4.7")
 addSbtPlugin("org.scalameta" % "sbt-mdoc" % "2.7.2")
 addSbtPlugin("com.lightbend.sbt" % "sbt-java-formatter" % "0.8.0")

--- a/tests/shared/src/test/scala/cats/effect/IOFiberSuite.scala
+++ b/tests/shared/src/test/scala/cats/effect/IOFiberSuite.scala
@@ -33,7 +33,7 @@ class IOFiberSuite extends BaseSuite with DetectPlatform {
         s <- IO(f.toString)
         // _ <- IO.println(s)
         _ <- f.cancel
-        _ <- IO(assert(s.matches(pattern)))
+        _ <- IO(assert(s.matches(pattern), s))
       } yield ()
     }
 
@@ -48,7 +48,7 @@ class IOFiberSuite extends BaseSuite with DetectPlatform {
         _ <- IO.sleep(100.milli)
         s <- IO(f.toString)
         _ <- f.cancel
-        _ <- IO(assert(s.matches(pattern)))
+        _ <- IO(assert(s.matches(pattern), s))
       } yield ()
     }
   } else {

